Core/include/Acts/Propagator/DirectNavigator.hpp

```diff
@@ -63,7 +65,7 @@ class DirectNavigator {
     Options options;
 
     /// Index of the next surface to try
-    std::size_t surfaceIndex = 0;
+    int surfaceIndex = 0;
```

Comment:
Why no `std::size_t`?

Reply:
I need a signed type because the index can now go below `0`